### ISSUE TITLE
Trying to refine poses fails

### ISSUE THREAD

**Issue by Guillawme** [zhonge/cryodrgn#43]
Hello,

I am trying to run a training with pose refinement, but it fails before it even starts. Here are the command I ran and the error I got:

```bash
cryodrgn train_vae j167-particles-box128_for-cryodrgn.mrcs --o train_009/ --zdim 8 --poses j167-particles-poses-from-RELION.pkl --ctf j167-particles-CTFs-from-RELION.pkl --seed 12345 -n 50 --domain hartley --do-pose-sgd
```
I am using `--domain hartley` because an initial attempt without this option told me it is necessary for pose refinement.

```python
2021-03-15 16:25:28     /opt/miniconda3/envs/cryodrgn-0.3.1/bin/cryodrgn train_vae j167-particles-box128_for-cryodrgn.mrcs --o train_009/ --zdim 8 --poses j167-particles-poses-from-RELION.pkl --ctf j167-particles-CTFs-from-RELION.pkl --seed 12345 -n 50 --domain hartley --do-pose-sgd
2021-03-15 16:25:28     Namespace(amp=False, batch_size=8, beta=1.0, beta_control=None, checkpoint=1, ctf='/data/luka/processing/201208_relion/cryodrgn/j167-box128-exploration/j167-particles-CTFs-from-RELION.pkl', datadir=None, do_pose_sgd=True, domain='hartley', emb_type='quat', enc_mask=None, encode_mode='resid', func=<function main at 0x7f1dffde65f0>, ind=None, invert_data=True, lazy=False, load=None, log_interval=1000, lr=0.0001, multigpu=False, norm=None, num_epochs=50, outdir='/data/luka/processing/201208_relion/cryodrgn/j167-box128-exploration/train_009', particles='/data/luka/processing/201208_relion/cryodrgn/j167-box128-exploration/j167-particles-box128_for-cryodrgn.mrcs', pdim=256, pe_dim=None, pe_type='geom_lowf', players=3, pose_lr=0.0003, poses='/data/luka/processing/201208_relion/cryodrgn/j167-box128-exploration/j167-particles-poses-from-RELION.pkl', pretrain=1, qdim=256, qlayers=3, relion31=False, seed=12345, tilt=None, tilt_deg=45, use_real=False, verbose=False, wd=0, window=True, zdim=8)
2021-03-15 16:25:28     Use cuda True
2021-03-15 16:25:29     Loaded 43698 128x128 images
2021-03-15 16:25:55     Normalized HT by 0 +/- 231.78614807128906
Traceback (most recent call last):
  File "/opt/miniconda3/envs/cryodrgn-0.3.1/bin/cryodrgn", line 33, in <module>
    sys.exit(load_entry_point('cryodrgn==0.3.1', 'console_scripts', 'cryodrgn')())
  File "/opt/miniconda3/envs/cryodrgn-0.3.1/lib/python3.7/site-packages/cryodrgn-0.3.1-py3.7.egg/cryodrgn/__main__.py", line 50, in main
    args.func(args)
  File "/opt/miniconda3/envs/cryodrgn-0.3.1/lib/python3.7/site-packages/cryodrgn-0.3.1-py3.7.egg/cryodrgn/commands/train_vae.py", line 318, in main
    pose_optimizer = torch.optim.SparseAdam(posetracker.parameters(), lr=args.pose_lr) if do_pose_sgd else None
  File "/opt/miniconda3/envs/cryodrgn-0.3.1/lib/python3.7/site-packages/torch/optim/sparse_adam.py", line 49, in __init__
    super(SparseAdam, self).__init__(params, defaults)
  File "/opt/miniconda3/envs/cryodrgn-0.3.1/lib/python3.7/site-packages/torch/optim/optimizer.py", line 47, in __init__
    raise ValueError("optimizer got an empty parameter list")
ValueError: optimizer got an empty parameter list
```

Is there anything wrong with my command? I am keeping most options to their defaults values, and the exact same command without `--domain hartley --do-pose-sgd` completes successfully, so not sure what I'm missing.

Thank you in advance for your help.

**Comment by zhonge** [comment]
Can you provide the version of pytorch and cryodrgn? I suspect it is this incompatibility with pytorch1.7+ documented here: https://github.com/zhonge/cryodrgn/issues/29

It is fixed in the latest version of the software (`git pull origin master`), but I probably shouldn't have closed the issue until I released that version.

**Comment by Guillawme** [comment]
Ah sorry, I should have also checked closed issues before posting.

I have cryodrgn version 0.3.1 (as in, the git tag pointing to commit 5581638) and pytorch version 1.7.1 according to `conda list`, so most likely it's the same problem. I will try updating to the latest commit on the master branch and see if it fixes it. Is master stable enough for regular use, or do you recommend I make another conda environment to test this fix?

**Comment by Guillawme** [comment]
I confirm that refining poses runs fine with commit 7ca6e4c (current tip of master branch), so this can be closed when 0.3.2 is released.
